src/transaction/components/TransactionButton.tsx

```diff
@@ -30,61 +34,106 @@ export function TransactionButton({
   const accountChainId = chainId ?? useChainId();
   const { showCallsStatus } = useShowCallsStatus();
 
+  const isLegacyTransactionInProgress =
+    lifecycleStatus.statusName === 'transactionLegacyExecuted' &&
+    transactionCount !==
+      lifecycleStatus?.statusData?.transactionHashList?.length;
+
   const isInProgress =
     lifecycleStatus.statusName === 'buildingTransaction' ||
     lifecycleStatus.statusName === 'transactionPending' ||
+    isLegacyTransactionInProgress ||
     isLoading;
+
   const isMissingProps = !transactions || !address;
   const isWaitingForReceipt = !!transactionId || !!transactionHash;
 
   const isDisabled =
     !receipt &&
     (isInProgress || isMissingProps || isWaitingForReceipt || disabled);
 
-  const displaySpinner = isSpinnerDisplayed({
+  const displayPendingState = isSpinnerDisplayed({
     errorMessage,
     hasReceipt: !!receipt,
-    isLoading,
-    lifecycleStatus,
+    isInProgress,
     transactionHash,
     transactionId,
   });
 
-  const buttonContent = useMemo(() => {
+  const { errorText, successText, pendingText } = useMemo(() => {
+    const successText = successOverride?.text
+      ? successOverride?.text
+      : 'View transaction';
+
+    const errorText = errorOverride?.text ? errorOverride?.text : 'Try again';
+
+    const pendingText = pendingOverride?.text;
+
+    return { successText, errorText, pendingText };
+  }, [errorOverride, pendingOverride, successOverride]);
```

Comment:
you should be able to simplify these to const errorText = errorOverride?.text ?? 'Try again', you could also set pendingText = pendingOverride?.text ?? <Spinner/> which would probably simplify render some.

Reply:
not sure what you mean about the `pendingText` line but i updated the others

Reply:
if you had const pendingText = pendingOverride?.text ?? <Spinner/> i think you'd be able to remove to. remove [this logic](https://github.com/coinbase/onchainkit/pull/1460/files#diff-bb9ce23f9b647b4d1b981336417d591dcf4aae8ac2b3d60757fe8f80f2ad43a1R151-R158)  and just display buttonContent always

Reply:
ohh i see yeah i didn't do that because buttonContent is rendered inside this `span` element so i'd have to have `buttonContent` return `successText`, `errorText` and `idleText` within a `span`. do you think that is cleaner?
```
      {displayPendingState && !pendingText ? (
        <Spinner />
      ) : (
        <span
          className={cn(text.headline, color.inverse, 'flex justify-center')}
        >
          {buttonContent}
        </span>
      )}
```